File: iot/CryptoTerm-main/CryptoTerm/Services/DeviceServices/Entities/DeviceResponse.swift
import Foundation

struct DeviceResponse: Decodable {
    let id: String
    let name: String
    let ipAddress: String
    let registeredAt: Date
    let minerIds: [String]

    enum CodingKeys: String, CodingKey {
        case id = "id"
        case name = "name"
        case ipAddress = "ip_address"
        case registeredAt = "registered_at"
        case minerIds = "miner_ids"
    }
}

File: iot/cryptoterm_backend-refactor/src/main/java/com/cryptoterm/backend/device/presentation/dto/DeviceResponse.java
package com.cryptoterm.backend.dto;

import com.fasterxml.jackson.annotation.JsonProperty;
import java.time.OffsetDateTime;
import java.util.List;
import java.util.UUID;

public record DeviceResponse(
    UUID id,
    String name,
    @JsonProperty("ip_address")
    String ipAddress,
    @JsonProperty("registered_at")
    OffsetDateTime registeredAt,
    @JsonProperty("miner_ids")
    List<UUID> minerIds
) {}

